resources/docs/livewire/showing-chirps.md

```diff
@@ -0,0 +1,313 @@
+<?php
+// [tl! collapse:start]
+use Livewire\Attributes\Rule;
+use Livewire\Volt\Component;
+
+new class extends Component
+{
+    #[Rule('required|string|max:255')]
+    public $message = '';
+    // [tl! collapse:end]
+    public function store(): void
+    {
+        $this->validate();
+
+        auth()->user()->chirps()->create([
+            'message' => $this->message,
+        ]);
+
+        $this->message = '';
+        // [tl! add:start]
+        $this->dispatch('chirp-created'); // [tl! add:end]
+    }
+} ?> <!-- [tl! collapse:start] -->
+
+<div> <!-- [tl! collapse:start] -->
+    <form wire:submit="store">
+        <textarea
+            wire:model="message"
+            placeholder="{{ __('What\'s on your mind?') }}"
+            class="block w-full border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm"
+        />
+        <x-input-error :messages="$errors->get('message')" class="mt-2" />
+        <x-primary-button class="mt-4">{{ __('Chirp') }}</x-primary-button>
+    </form> <!-- [tl! collapse:end] -->
+</div>
+```
+
+```php tab=Functional filename=resources/views/livewire/chirps/create.blade.php
+<?php
+// [tl! collapse:start]
+use function Livewire\Volt\{state};
+use function Livewire\Volt\{rules, state};
+
+state(['message' => '']);
+
+rules(['message' => 'required|string|max:255']);
+// [tl! collapse:end]
+$store = function () {
+    $this->validate();
+
+    auth()->user()->chirps()->create([
+        'message' => $this->message,
+    ]);
+
+    $this->message = '';
+    // [tl! add:start]
+    $this->dispatch('chirp-created'); // [tl! add:end]
+};
+
+?>
+
+<div> <!-- [tl! collapse:start] -->
+    <form wire:submit="store">
+        <textarea
+            wire:model="message"
+            placeholder="{{ __('What\'s on your mind?') }}"
+            class="block w-full border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm"
+        />
+
+        <x-input-error :messages="$errors->get('message')" class="mt-2" />
+        <x-primary-button class="mt-4">{{ __('Chirp') }}</x-primary-button>
+    </form> <!-- [tl! collapse:end] -->
+</div>
+```
+
+Now, we can update our `chirps.list` component to listen for the `chirp-created` event, and update the list of Chirps:
+
+```php tab=Class filename=resources/views/livewire/chirps/list.blade.php
+<?php
+
+use App\Models\Chirp;
+use Illuminate\Database\Eloquent\Collection;
+use Livewire\Attributes\On; // [tl! add]
+use Livewire\Volt\Component;
+
+new class extends Component
+{
+    public Collection $chirps;
+
+    public function mount(): void
+    {
+        $this->chirps = Chirp::with('user') // [tl! remove:start]
+            ->latest()
+            ->get(); // [tl! remove:end]
+        $this->getChirps(); // [tl! add]
+    }
+    // [tl! add:start]
+    #[On('chirp-created')]
+    public function getChirps()
```

Comment:
Should this have a `void` return type to be consistent with the other class methods? It might be missing in a few other places too.

Reply:
Fixed.